app/controllers/name_controller/create_and_edit_name.rb

```diff
@@ -128,8 +129,11 @@ def merge_names(new_name)
 
   def perform_merge_names(new_name)
```

Comment:
**Do you a 10,000 ft. view how to refactor `perform_merge_names`?** 
(It badly needs refactoring. I've tried a few times, but always break the tests. I may try again tomorrow.)
**Would it make sense to move it to a separate file because of this offense: `(Metrics/ClassLength: Class has too many lines. [263/250]`?**

Reply:
While it's well-known that I have a far greater tolerance for long and/or complex methods and files (and therefore think it's fine the way it is -- "if it ain't broke don't fix it"), I accept that others feel differently.

The short answer, I think, is to notice that there are three comments in perform_merge_names: if it needs a comment, maybe the bits of code that require a comment should be moved into their own methods.  Hell you could even automate it by turning the comment into a method name!  And yeah, move the methods that are exclusively used by the merge to a new file.

But yuck, we already started that process with "set_locked_citation_and_notes".  Already I have to jump around the code to understand how this method works.  At least in that case, that trivial method is used in two other places as well, so it satisfies a DRY itch.

No, I think the long answer involves recognizing the many bad smells coming off of this code and therefore realizing that a major refactor is probably required to make it more natural and elegant.  And that means expanding the realm of refactor to include Name and revisiting where various logic belongs.  We have a long history of randomly throwing logic in wherever we happen to be working, without asking ourselves "where does this really belong?"  Which logic belongs in the controller?  Which in the model?  (View isn't a concern in this case at least.)

So for example, the actual merge, as I'd call it, is buried hidden toward the bottom of perform_merge_names -- "new_name.merge(@name)".  That's a model method.  Why aren't things like deprecation status, author, order, etc. also handled in the model?  The superficial reasoning is clear enough: everything requiring access to the parser or params is being handled by the controller, the rest is left for the model.  But in all honesty, this action doesn't actually require -- or shouldn't! -- either.

Maybe, therefore, the "earliest" problem is simply that there is no dedicated request for merging two names, and instead is handled as a screwy side-effect of changing the name of an existing name!  (I admit full responsibility for this state of affairs.)  Then, that removes the parameters and parser BS in this perform_name_merge method, allowing most of this to move into the model where (it seems to me) it really belongs.  When I think about the general problem -- how do you merge two names? -- I can't think of any compelling reason there should be any user options!  Either it can happen or it can't.  Let the model decide which to keep and which to delete, which data to keep in the final merged name, and so on.

But is this really what this PR is about?  Of course not.  We're talking about a *major* refactor at this point, none of which has anything to do with Mycobank identifiers.

So I repeat: "don't fix what ain't broke".  Of course, that's just my opinion on the subject.

Reply:
In case you decide you want to refactor it anyway, it strikes me that there there are three sections to this code.  1) the lines before the potential swap, 2) the swap, and 3) the cleanup afterwords.  Reading the code, it's not clear to me why there is so much happening before the potential swap and if that is really the expected behavior.  If I were going after this method I would try to closely follow the process outlined in Fowler & Beck's book "Refactoring" (https://martinfowler.com/books/refactoring.html) where you make really small changes and then verify that the tests continue to pass.  I might start with seeing if you can move the swap logic earlier.

Reply:
Thank you both for the comments and the 10,000' views.
It's not broken in the sense that all the tests pass. But it's quite complicated -- it took me multiple hours and re-reads, and some experimentation to understand it to some extent. And I made it worse.
But my current thinking is to make the refactor a separate, follow-up story. 
In addition to the ideas that each of you suggest, it would help me to change/add more expressive variable names, e.g., things like `surviving name` `name_to_be_merged`, name_being_edited` `found_oame`